app/src/main/java/org/breezyweather/remoteviews/config/AbstractWidgetConfigActivity.kt

```diff
@@ -192,19 +195,18 @@ abstract class AbstractWidgetConfigActivity : GeoActivity() {
         KeyboardResizeBugWorkaround.assistActivity(this)
     }
 
-    @Deprecated("Deprecated in Java")
-    override fun onBackPressed() {
-        if (mBottomSheetBehavior!!.state == BottomSheetBehavior.STATE_EXPANDED) {
-            setBottomSheetState(true)
-            return
-        }
-        val time = System.currentTimeMillis()
-        if (time - mLastBackPressedTime < 2000) {
-            super.onBackPressed()
-            return
+    private val onBackPressedCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() {
+            if (mBottomSheetBehavior!!.state == BottomSheetBehavior.STATE_EXPANDED) {
+                setBottomSheetState(true)
+            } else if (System.currentTimeMillis() - mLastBackPressedTime < 3650) {
+                isEnabled = false
```

Comment:
`isEnabled` controls the [enabled state](https://developer.android.com/reference/androidx/activity/OnBackPressedCallback#getIsEnabled()) of `OnBackPressedCallback`. As long as this is `true`, the custom `handleOnBackPressed` is used.

By setting this to `false` and calling `onBackPressedDispatcher.onBackPressed()` afterwards, it will effectively trigger a default back press.